codyze-backends/cpg/src/test/kotlin/de/fraunhofer/aisec/codyze/backends/cpg/NfaDfaConstructionTest.kt

```diff
@@ -208,17 +208,15 @@ class NfaDfaConstructionTest {
             // cm.reset()?
             val dfa = orderExpressionToDfa {
                 +TestClass::init
-                +some {
-                    +(
-                        TestClass::create or
-                            group {
-                                +TestClass::start
-                                +TestClass::process.maybe()
-                                +TestClass::finish
-                            }
-                        )
+                some {
+                    TestClass::create or
+                        group {
+                            +TestClass::start
+                            TestClass::process.maybe()
```

Comment:
Having to use '+' on standalone `OrderTokens` but not on `OrderTokens` on which we call something e.g., `TestClass::process.maybe()` might be confusing. Perhaps we should change it to be `maybe(TestClass::process)`.
Then the behavior would be more predictable:
- no need for '+' when calling any method
- add a '+' for standalone `OrderTokens`

Reply:
Additionally, we could make the `add` method public. Then the user would not have to use '+' anywhere, but still has the option to do so.
Having the `add` method publicly available should also help with intellij autocomplete.

Reply:
> Having to use '+' on standalone `OrderTokens` but not on `OrderTokens` on which we call something e.g., `TestClass::process.maybe()` might be confusing. Perhaps we should change it to be `maybe(TestClass::process)`. Then the behavior would be more predictable:
> 
>     * no need for '+' when calling any method
> 
>     * add a '+' for standalone `OrderTokens`

That is true, the suggestions sounds good! We already have `maybe(vararg tokens)` as a function which covers `maybe(TestClass::process)` so I will just delete `OrderToken.maybe()` and the corresponding functions for the other quantifiers.